dll/win32/shell32/CUserNotification.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 Hermes Belusca-Maito
+ *
+ * Pass on icon notification messages to the systray implementation
+ * in the currently running shell.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+#ifndef _USERNOTIFICATION_H_
+#define _USERNOTIFICATION_H_
+
+class CUserNotification :
+    public CComCoClass<CUserNotification, &CLSID_UserNotification>,
+    public CComObjectRootEx<CComMultiThreadModelNoCS>,
+    public IUserNotification
+//  public IUserNotification2   // On Vista+
+{
+private:
+    /* Private stuff... */
+
+public:
+    CUserNotification();
+    ~CUserNotification();
+
+    // IUserNotification
+    virtual HRESULT STDMETHODCALLTYPE SetBalloonInfo(
+        IN LPCWSTR pszTitle,
+        IN LPCWSTR pszText,
+        IN DWORD dwInfoFlags);
+
+    virtual HRESULT STDMETHODCALLTYPE SetBalloonRetry(
+        IN DWORD dwShowTime,  // Time in milliseconds
+        IN DWORD dwInterval,
+        IN UINT cRetryCount);
+
+    virtual HRESULT STDMETHODCALLTYPE SetIconInfo(
+        IN HICON hIcon,
+        IN LPCWSTR pszToolTip);
+
+    virtual HRESULT STDMETHODCALLTYPE Show(
+        IN IQueryContinue *pqc,
+        IN DWORD dwContinuePollInterval);
+
+    virtual HRESULT STDMETHODCALLTYPE PlaySound(
+        IN LPCWSTR pszSoundName);
+
+#if 0
+    // IUserNotification2
+    virtual HRESULT STDMETHODCALLTYPE Show(
+        IN IQueryContinue *pqc,
+        IN DWORD dwContinuePollInterval,
+        IN IUserNotificationCallback *pSink);
+#endif
+
+DECLARE_REGISTRY_RESOURCEID(IDR_USERNOTIFICATION)
```

Comment:
Please indent this (and lines below) with 1 tab (4 spaces)

Reply:
thkx will do it.